### PR TITLE
Complete 112. Path Sum

### PR DESCRIPTION
**Pull request by updownlife**

### CODE REVIEW

File: hudeven/P112.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        if root:
+            sum -= root.val
+            if sum == 0 and self.isLeaf(root):
+                return True;
+            return self.hasPathSum(root.left, sum) or self.hasPathSum(root.right, sum)
+        else:
+            return False
+
+    def isLeaf(self, node):
+        return not node.left and not node.right
```
Comment (lionelfeng) [inline hudeven/P112.py]: I like it!

Comment (lionelfeng) [inline hudeven/P112.py]: why would you need a separated one-line function if it was only called once. But it's more readable indeed.

Comment (updownlife) [inline hudeven/P112.py]: yeah, just to make it more readable. Do you think python interpreter would optimize it just like C++ optimizing inline function?

Comment (lionelfeng) [inline hudeven/P112.py]: not sure about the optimization... teach me!!!

Comment (updownlife) [inline hudeven/P112.py]: It won't! Python will not do inline optimization for short functions!
http://stackoverflow.com/a/6442497/1954090

**Comment by lionelfeng** [comment]
lgtm

**Pull request merged**